### PR TITLE
Implement manage stories filter

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link

[Implement manage stories filter](https://www.notion.so/uwblueprintexecs/Implement-manage-stories-filter-8b37465fa9e04a3ea89cf136345e7398)

## Implementation description

- allow optional story title, start/end date upload filtering in get stories query
- add date filter to manage stories table

## Steps to test

1. Rebuild the containers 
2. Update the upload date of a few stories `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; UPDATE stories SET date_uploaded="2021-12-21 06:48:42" where id IN (4,5,6);"`; `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; UPDATE stories SET date_uploaded="2021-12-08 06:48:42" where id IN (1,2);"`; 
3. Login in as an admin
4. Navigate to the [manage stories tab](http://localhost:3000/#/?tab=1)
5. Play around with the date filter: if just one date is selected, stories with upload date >= than it should be returned; if a date range is selected, stories with an upload date within that selected time frame should be returned
6. Verify that filters on other tables still work as expected

manage story date

## What should reviewers focus on?

- Is there a better way to style the date picker?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/TableFilter.tsx [hunk 1]
```diff
@@ -13,27 +13,34 @@ import {
...
 } from "@chakra-ui/react";
+import DatePicker from "react-datepicker";
 import { languageOptions } from "../../constants/Languages";
 import { levelOptions } from "../../constants/Levels";
 import { stageOptions } from "../../constants/Stage";
 import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
 import { FILTER_TOOL_TIP_COPY } from "../../utils/Copy";
 import DropdownIndicator from "../utils/DropdownIndicator";
 import { colourStyles } from "../../theme/components/Select";
+// import StyledDatePicker from "./StyledDatePicker";
```
Comment (gaoxk) [inline frontend/src/components/admin/TableFilter.tsx:24]: nit: remove

File: frontend/src/theme/components/DatePicker.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
+@import "node_modules/react-datepicker/src/stylesheets/datepicker.scss";
```
Comment (gaoxk) [inline frontend/src/theme/components/DatePicker.scss:1]: woah, is this file necessary to use the DatePicker?

File: frontend/src/components/auth/ResetPassword.tsx [hunk 3]
```diff
@@ -49,6 +49,7 @@ const ResetPassword = ({ email }: ResetPasswordProps) => {
       if (typeof err === "string") {
         handleErrorOnReset(err);
       } else {
+        // eslint-disable-next-line no-console
         console.log(err);
```
Comment (gaoxk) [inline frontend/src/components/auth/ResetPassword.tsx:53]: oh shoot, we should really window.alert the errs so the user can see them. I'll make a note to swap them all in the code base
where you dealing with eslint errors? Is that why theses have been added in?

**Review by gaoxk — CHANGES_REQUESTED** [review]
hmmmmm maybe im not testing this correctly, but nothing i do to the filters makes the table change :0 I took a look at the network and all the gql requests are always returning the full list of stories

File: frontend/src/components/auth/ResetPassword.tsx (hunk 3 above)
Comment (jennchenn) [reply frontend/src/components/auth/ResetPassword.tsx:53]: Sorry should've added a comment about this; the firebase build kept failing with an error saying `Treating warnings as errors because process.env.CI = true`
https://github.com/uwblueprint/planet-read/runs/4676380763?check_suite_focus=true

File: frontend/src/theme/components/DatePicker.scss (hunk 2 above)
Comment (jennchenn) [reply frontend/src/theme/components/DatePicker.scss:1]: It looks really strange without it: 
image

**Comment by jennchenn** [comment]
> hmmmmm maybe im not testing this correctly, but nothing i do to the filters makes the table change :0 I took a look at the network and all the gql requests are always returning the full list of stories

Made a silly mistake while rebasing 😞 if you try again now they should hopefully work!

File: frontend/src/theme/components/DatePicker.scss (hunk 2 above)
Comment (gaoxk) [reply frontend/src/theme/components/DatePicker.scss:1]: wowsa, that is funky. How did you figure that this file was needed?

File: frontend/src/components/review/CommentsPanel.tsx [hunk 4]
```diff
@@ -115,25 +115,28 @@ const CommentsPanel = ({
...
-                  translatorId={translatorId}
-                  translatorName={translatorName}
-                />
-              );
-            }
-            return null;
-          })}
+          {
+            // eslint-disable-next-line func-names
+            comments.map(function (comment: CommentResponse, i: number) {
```
Comment (gaoxk) [inline frontend/src/components/review/CommentsPanel.tsx:120]: hmmm would this error disappear if the function were an arrow function? eg `(comment: CommentResponse, i: number) 
 => {....`

**Review by gaoxk — APPROVED** [review]
worksssss,nice!

File: frontend/src/components/review/CommentsPanel.tsx (hunk 4 above)
Comment (jennchenn) [reply frontend/src/components/review/CommentsPanel.tsx:120]: yes this seems to work: `{comments.map((comment: CommentResponse, i: number) => {` --would you prefer this?

Comment (gaoxk) [reply frontend/src/components/review/CommentsPanel.tsx:120]: yeah lets do it!

**Comment by gaoxk** [comment]
```
src/components/admin/StoriesTable.tsx
  Line 79:7:  Unexpected alert  no-alert
```

**Comment by gaoxk** [comment]
https://github.com/firebase/firebase-tools/issues/3999 

fyi ongoing firebase deployment issue, dont worry too much about resolving it in this pr if its too much
